client/sites/settings/controller.tsx

```diff
@@ -1,47 +1,42 @@
 import { __ } from '@wordpress/i18n';
-import makeSidebar, { PanelWithSidebar } from '../components/panel-sidebar';
+import { useSelector } from 'react-redux';
+import isSiteWpcomStaging from 'calypso/state/selectors/is-site-wpcom-staging';
+import { useSelectedSiteSelector } from 'calypso/state/sites/hooks';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import { SidebarItem, Sidebar, PanelWithSidebar } from '../components/panel-sidebar';
+import AdministrationSettings from './administration';
+import AgencySettings from './agency';
+import CachesSettings from './caches';
+import SiteSettings from './site';
+import WebServerSettings from './web-server';
 import type { Context as PageJSContext } from '@automattic/calypso-router';
 
-const SettingsSidebar = makeSidebar( {
-	items: [
-		{
-			key: 'site',
-			get label() {
-				return __( 'Site' );
-			},
-		},
-		{
-			key: 'administration',
-			get label() {
-				return __( 'Administration' );
-			},
-		},
-		{
-			key: 'agency',
-			get label() {
-				return __( 'Agency' );
-			},
-		},
-		{
-			key: 'caches',
-			get label() {
-				return __( 'Caches' );
-			},
-		},
-		{
-			key: 'web-server',
-			get label() {
-				return __( 'Web Server' );
-			},
-		},
-	],
-} );
+export function SettingsSidebar() {
```

Comment:
Okay, I'm now convinced that this is a better approach than `makeSidebar()`. The only downside is having to list down all the paths, duplicating what we have in `index.tsx`. But I guess that's fine for now. Thank you!

Reply:
I personally prefer this because it's grepable but if you really want it we can abstract it into path functions similar to how domains works: https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/domains/paths.js#L111

Reply:
> it's grepable

That's interesting! I never thought about it.

> if you really want it we can abstract it

I don't think it's necessary for now, it's very low priority 😄